### PR TITLE
docs: refactor Text documentation and add missing guidance for presets/alignment

### LINKED ISSUE #24341: `weight` attribute does not work with predefined styled text components
### Environment Information

- **Package version(s)**: `@fluent/react-components`==9.2.0
- **Browser and OS versions**: _(irrelevant)_

### Describe the issue:

`weight` attribute does not work with predefined styled text components (`<Body1 />`, `<Caption1 />`, `<Caption2 />`, etc.)

#### Actual behavior:

```tsx
import { Text, Body1 } from "@fluentui/react-components";
import * as React from "react";
export const Weight = () => (
  <div
    style={{
      display: "flex",
      gap: 20
    }}
  >
    <Text weight="semibold">Semibold weight 1</Text>
    <Body1 weight="semibold">Semibold weight 2</Body1>
  </div>
);
```

"Semibold weight 2" is rendered in regular weight.

#### Expected behavior:

Both texts should be in semibold weight.

### If applicable, please provide a codepen repro:

https://codesandbox.io/s/mystifying-gwen-yyr4h1?file=/example.tsx:0-307

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Low

Products/sites affected: MS internal-only product

### LINKED ISSUE #25548: [Bug]: Text wrappers lack proper documentation explaining usage and motivation
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
n/a documentation bug
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-text--default

## Actual Behavior
Text documentation shows the wrappers and their styles, but does not explain the concept/purpose of the wrappers, nor show much documentation on the usage. 

## Expected Behavior
The wrappers with fixed styles is a new concept, we should explain it, and explain it's motivation. 

See #24341 for issue filed on this.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by marcosmoura**
## Changed
This PR includes a refactor of all stories related to Text component.
- Add missing documentation for Text presets
- Add missing documentation for Text alignment props
- Add better examples and more descriptive texts
- Simplification of styles using Griffel
- Improve layout of examples

## Related Issue(s)
Fixes #24341
Fixes #25548

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/Avatar-mount_base.html">1251</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/Avatar-mount.html">1282</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/Button-mount_base.html">927</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/Button-mount.html">931</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProvider-mount_base.html">1491</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProvider-mount.html">1472</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-mount_base.html">585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-mount.html">576</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">541</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">548</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">573</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/MakeStyles-mount_base.html">1956</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/MakeStyles-mount.html">1946</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/SpinButton-mount_base.html">2372</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25587/perf-test-react-components/SpinButton-mount.html">2352</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.941 kB`<br />`17.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.948 kB`<br />`52.993 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `24.112 kB`<br />`7.892 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |   `44.42 kB`<br />`12.815 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `11.828 kB`<br />`4.627 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `15.138 kB`<br />`5.064 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/fa97592b4f9020e187353c781766918e37be8d40'>fa97592b4f9020e187353c781766918e37be8d40</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+    width: '240px',
+    display: 'block',
+  },
+});
+
+export const Truncate = () => {
+  const styles = useStyles();
+
+  return (
+    <Text truncate wrap={false} className={styles.text}>
```
Comment (Hotell) [inline packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx:16]: what's the wrap=false for ?

**Review by Hotell — COMMENTED** [review]
I like the restructuring which is more aligned with design spec 🙌,

added some notes that should be resolved

File: packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx (hunk 1 above)
Comment (marcosmoura) [reply packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx:16]: The `wrap={false}` will include the `whitespace: nowrap` and `overflow: hidden` to the styles.
image
image

File: packages/react-components/react-text/stories/Text/TextDescription.md [hunk 2]
```diff
@@ -1 +1,13 @@
 Typography and styling abstraction component used to ensure consistency and standardize text throughout your application.
+
```
[the remark this answers is not in the document]
Comment (marcosmoura) [reply packages/react-components/react-text/stories/Text/TextDescription.md:2]: Fair! Updated!

**Comment by marcosmoura** [comment]
> I like the restructuring which is more aligned with design spec 🙌,
> 
> added some notes that should be resolved
Made some changes to the code and added a reply on the `wrap` usage.

**Review by Hotell — APPROVED** [review]

File: packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx (hunk 1 above)
Comment (Hotell) [reply packages/react-components/react-text/stories/Text/TextTruncate.stories.tsx:16]: gotcha ty

**Pull request merged**